cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/AbstractSchedulerService.java

```diff
@@ -254,6 +286,18 @@ private Scheduler getSchedulerForSchedule(ProgramId program, String scheduleName
     return getScheduler(schedule);
   }
 
+  @Nullable
+  private Scheduler getScheduler(ProgramSchedule schedule) {
+    if (schedule.getTrigger() instanceof TimeTrigger) {
+      return timeScheduler;
+    } else if (schedule.getTrigger() instanceof StreamSizeTrigger) {
+      return streamSizeScheduler;
+    } else {
+      // Return null for unhandled type of ProgramSchedule
```

Comment:
shouldn't we throw an exception in that case? It can only happen if the code has a bug, right?
At least LOG.warn?